src/dreyfus_index.erl

```diff
@@ -222,16 +210,32 @@ code_change(_OldVsn, State, _Extra) ->
 
 % private functions.
 
+reply_or_update(IndexPid, Index, State) ->
+    case reply_with_index(IndexPid, Index, State#state.waiting_list) of
+        [] ->
+            State#state{
+                updater_pid=nil,
+                waiting_list=[]
+            };
+        StillWaiting ->
+            Pid = spawn_link(fun() -> dreyfus_index_updater:update(IndexPid, Index) end),
+            State#state{
+                updater_pid=Pid,
+                waiting_list=StillWaiting
+            }
+    end.
+
 open_index(DbName, #index{analyzer=Analyzer, sig=Sig}) ->
     Path = <<DbName/binary,"/",Sig/binary>>,
     case clouseau_rpc:open_index(self(), Path, Analyzer) of
         {ok, Pid} ->
-            case clouseau_rpc:get_update_seq(Pid) of
-                {ok, Seq} ->
-                    {ok, Pid, Seq};
-                Error ->
-                    Error
-            end;
+            UpdateSeq = case clouseau_rpc:get_update_seq(Pid) of
```

Comment:
There are some changes here before, but they are removed.
I will change it back to origin status.